manifest/v1alpha/budgetadjustment/validation.go

```diff
@@ -0,0 +1,89 @@
+package budgetadjustment
+
+import (
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/teambition/rrule-go"
+
+	validationV1Alpha "github.com/nobl9/nobl9-go/internal/manifest/v1alpha"
+	"github.com/nobl9/nobl9-go/internal/validation"
+	"github.com/nobl9/nobl9-go/manifest/v1alpha"
+)
+
+var budgetAdjustmentValidation = validation.New[BudgetAdjustment](
+	validation.For(func(b BudgetAdjustment) Metadata { return b.Metadata }).
+		Include(metadataValidation),
+	validation.For(func(b BudgetAdjustment) Spec { return b.Spec }).
+		WithName("spec").
+		Include(specValidation),
+)
+
+var metadataValidation = validation.New[Metadata](
+	validationV1Alpha.FieldRuleMetadataName(func(m Metadata) string { return m.Name }),
+	validationV1Alpha.FieldRuleMetadataDisplayName(func(m Metadata) string { return m.DisplayName }),
+)
+
+var specValidation = validation.New[Spec](
+	validation.For(func(s Spec) string { return s.Description }).
+		WithName("description").
+		Rules(validation.StringDescription()),
+	validation.For(func(s Spec) time.Time { return s.FirstEventStart }).
+		WithName("firstEventStart").
+		Required(),
+	validation.For(func(s Spec) time.Duration { return s.Duration }).
+		WithName("duration").
+		Required().
+		Rules(durationValidationRule()),
+	validation.For(func(s Spec) string { return s.Rrule }).
+		WithName("rrule").
+		Rules(rruleValidationRule()),
```

Comment:
you can use `validation.Transform` to ensure this string is of type `rrule.Rule`:

```suggestion
	validation.Transform(func(s Spec) string { return s.Rrule }, rrule.StrToRRule).
		WithName("rrule"),
```